src/match.js

```diff
@@ -1,19 +1,25 @@
-var invoker = require('./invoker');
+var _curry2 = require('./internal/_curry2');
+var defaultTo = require('./defaultTo');
 
 
 /**
- * Tests a regular expression against a String
+ * Tests a regular expression against a String. Note that this function
+ * will return an empty array when there are no matches. This differs
+ * from the standard JavaScript string function which returns null when
+ * there are no matches.
  *
  * @func
  * @memberOf R
  * @category String
- * @sig RegExp -> String -> [String] | null
+ * @sig RegExp -> String -> [String]
```

Comment:
i think the best course is to document/warn about it, and recommend composing with `filter(complement(isNil))` (or wrapping in `Maybe`s)